test/app.test.js

```diff
@@ -0,0 +1,279 @@
+const webdriver = require('selenium-webdriver');
+const { until, By } = require('selenium-webdriver');
+const { Options } = require('selenium-webdriver/chrome');
+
+const assert = require('assert');
+
+const { server } = require('../src/js/mocks/server');
+
+const handler = require('serve-handler');
+const http = require('http');
+
+
+/*
+
+Test Auth functionalities (Login, Signup, Logout) with selenium and mocha
+
+*/
+
+
+describe('Authentication Testing', function() {
+	let driver;
+	let api_server;
+	let static_server;
+
+	const BASE_URL = `http://localhost:3000`;
+	const LOGIN_URL = `${BASE_URL}/accounts/login/`;
+	const SIGNUP_URL = `${BASE_URL}/accounts/register/`;
+	const CHANGE_PASSWORD_URL = `${BASE_URL}/account/change-password/`;
+
+	const VALID_USER = 'user';
+	const VALID_PASSWORD = 'pass';
+
+	before(async () => {
+		let options = new Options()
+		driver = new webdriver.Builder().forBrowser('chrome')
+		.setChromeOptions(options.headless())
+		.build();
+
+		api_server = server.listen(3004, () => {
+		  console.log('Mock Server is running')
+		})
+
+		static_server = http.createServer((request, response) => {
+			return handler(request, response, {
+				public: 'dist',
+				rewrites: [
+					{ "source": "accounts/login", "destination": "index.html" },
+					{ "source": "accounts/register", "destination": "index.html" },
+					{ "source": "account/change-password", "destination": "index.html" }
+				]
+			});
+		})
+
+		static_server.listen(3000, () => {
+		  console.log('Running at http://localhost:3000');
+		});
+	}, 50000);
+
+	after(async () => {
+	  await driver.quit();
+	  api_server.close(function () {
+	  	console.log('Mock Server shutdown')
+	  })
+
+	  static_server.close(function(){console.log('Shutting down static server')})
+	}, 50000);
+
+
+	describe('test for login functionality', function(){
+
+		let usernameBox;
+		let passwordBox;
+
+		before(async function(){
+			await driver.get(LOGIN_URL);
+			await driver.sleep(500);
+		});
+
+		beforeEach(async function() {
+			usernameBox = driver.findElement(By.id('my-muni-Username'));
+			passwordBox = driver.findElement(By.id('my-muni-Password'));
+
+			usernameBox.clear();
+			passwordBox.clear();
+		});
+
+		async function sendData(nm, p1){
+			usernameBox.sendKeys(nm);
+			passwordBox.sendKeys(p1);
+
+			const button = driver.findElement(By.className('button form-submit w-button'));
```

Comment:
you should scope selection of buttons to the modal because there might be more buttons like this hidden in the component library. It might be easier then to select using CSS selector rather than class name.